source/grid/grid_in.cc

```diff
@@ -2542,6 +2566,162 @@ void GridIn<dim, spacedim>::read_tecplot(std::istream &)
 }
 
 
+
+template <int dim, int spacedim>
+void GridIn<dim, spacedim>::read_assimp(const std::string &filename,
+                                        const unsigned int mesh_index,
+                                        const bool remove_duplicates,
+                                        const double tol,
+                                        const bool ignore_unsupported_types)
+{
+#ifdef DEAL_II_WITH_ASSIMP
+  // Only good for surface grids.
+  AssertThrow(dim<3, ExcImpossibleInDim(dim));
+
+  // Create an istance of the Importer class
+  Assimp::Importer importer;
+
+  // And have it read the given file with some  postprocessing
+  const aiScene *scene = importer.ReadFile( filename.c_str(),
```

Comment:
are you creating a memory leak here? Or is the memory managed by the importer object?

Reply:
It is managed by the importer object:

> Create an object of this class and call `ReadFile()` to import a file. If the import succeeds, the function returns a pointer to the imported data. The data remains property of the object, it is intended to be accessed read-only. The imported data will be destroyed along with the Importer object. If the import fails, ReadFile() returns a NULL pointer. In this case you can retrieve a human-readable error description be calling GetErrorString(). You can call ReadFile() multiple times with a single Importer instance. Actually, constructing Importer objects involves quite many allocations and may take some time, so it's better to reuse them as often as possible.